node/pkg/watchers/solana/client.go

```diff
@@ -548,6 +718,59 @@ func (s *SolanaWatcher) fetchMessageAccount(ctx context.Context, logger *zap.Log
 	return false
 }
 
+func (s *SolanaWatcher) processAccountSubscriptionData(ctx context.Context, logger *zap.Logger, data []byte) error {
+	// Do we have an error on the subscription?
+	var e EventSubscriptionError
+	err := json.Unmarshal(data, &e)
+	if err != nil {
+		logger.Error(*s.wsUrl, zap.Error(err))
+		p2p.DefaultRegistry.AddErrorCount(s.chainID, 1)
+		return err
+	}
+
+	if e.Error.Message != nil {
+		return errors.New(*e.Error.Message)
+	}
+
+	var res EventSubscriptionData
+	err = json.Unmarshal(data, &res)
+	if err != nil {
+		logger.Error(*s.wsUrl, zap.Error(err))
+		p2p.DefaultRegistry.AddErrorCount(s.chainID, 1)
+		return err
+	}
+
+	if res.Params == nil {
+		return nil
+	}
+
+	value := (*res.Params).Result.Value
+
+	if value.Account.Owner != s.rawContract {
+		// We got a message for the wrong contract on the websocket... uncomfortable...
+		return errors.New("Update for account with wrong owner")
+	}
+
+	data, err = base64.StdEncoding.DecodeString(value.Account.Data[0])
+	if err != nil {
+		logger.Error(*s.wsUrl, zap.Error(err))
+		p2p.DefaultRegistry.AddErrorCount(s.chainID, 1)
+		return err
+	}
+
+	// Other accounts owned by the wormhole contract seem to send updates...
+	switch string(data[:3]) {
+	case accountPrefixReliable:
```

Comment:
Go case blocks don't fall through by default so this is basically ignoring the `accountPrefixReliable` case.  Is that intentional?

If we actually do want to process the message in the `accountPrefixReliable` case then we either need to add an explicit fallthrough or check for both values in a single case:

```go
switch string(data[:3]) {
case accountPrefixReliable, accountPrefixUnreliable:
	// Handle message
}
```

If we don't want to process the message then we can remove that line completely.

Reply:
good catch